sdk/storage/azure_storage_blob/tests/blob_client.rs

```diff
@@ -15,348 +20,437 @@ use std::error::Error;
+    let (options, endpoint) = recorded_test_setup(recording).await;
     let container_name = recording
         .random_string::<17>(Some("container"))
         .to_ascii_lowercase();
     let blob_name = recording
         .random_string::<12>(Some("blob"))
         .to_ascii_lowercase();
 
+    let container_client_options = BlobContainerClientOptions {
+        client_options: options.clone(),
+        ..Default::default()
+    };
     // Act
-    let container_client = ContainerClient::new(
+    let container_client = BlobContainerClient::new(
         &endpoint,
         container_name.clone(),
         recording.credential(),
-        Some(options.clone()),
+        Some(container_client_options),
     )?;
     container_client.create_container(None).await?;
 
+    let blob_client_options = BlobClientOptions {
+        client_options: options.clone(),
+        ..Default::default()
+    };
     let blob_client = BlobClient::new(
         &endpoint,
         container_name,
         blob_name,
         recording.credential(),
-        Some(options),
+        Some(blob_client_options),
     )?;
 
     let block_1 = b"AAA";
     let block_2 = b"BBB";
     let block_3 = b"CCC";
 
+    let block_1_id: Vec<u8> = b"1".to_vec();
+    let block_2_id: Vec<u8> = b"2".to_vec();
+    let block_3_id: Vec<u8> = b"3".to_vec();
+
     blob_client
         .stage_block(
-            "1",
-            i64::try_from(block_1.len())?,
+            block_1_id.clone(),
+            u64::try_from(block_1.len())?,
             RequestContent::from(block_1.to_vec()),
             None,
         )
         .await?;
 
     blob_client
         .stage_block(
-            "2",
-            i64::try_from(block_2.len())?,
+            block_2_id.clone(),
+            u64::try_from(block_2.len())?,
             RequestContent::from(block_2.to_vec()),
             None,
         )
         .await?;
     blob_client
         .stage_block(
-            "3",
-            i64::try_from(block_3.len())?,
+            block_3_id.clone(),
+            u64::try_from(block_3.len())?,
             RequestContent::from(block_3.to_vec()),
             None,
         )
         .await?;
 
-    let latest_blocks: Vec<String> = vec![
-        base64::encode("1"),
-        base64::encode("2"),
-        base64::encode("3"),
-    ];
+    // Three Staged Blocks Scenario
+    let block_list = blob_client
+        .get_block_list(BlockListType::All, None)
+        .await?
+        .into_body()
+        .await?;
+
+    // Assert
+    assert_eq!(0, block_list.committed_blocks.len());
+    assert_eq!(3, block_list.uncommitted_blocks.len());
+
+    let latest_blocks: Vec<Vec<u8>> = vec![block_1_id, block_2_id, block_3_id];
 
     let block_lookup_list = BlockLookupList {
-        committed: None,
-        latest: Some(latest_blocks),
-        uncommitted: None,
+        committed: Vec::new(),
+        latest: latest_blocks,
+        uncommitted: Vec::new(),
```

Comment:
Does the "unused" ones not support `None` anymore? Sort of a shame to have to allocate empty vectors here.

Reply:
Nope, this is currently in alignment with the `BlockLookupList` model, which takes in `Vec<Vec<u8>> strictly (no `Option` here)

Reply:
@LarryOsterman convinced us - though not to hold up the MVP - that we should go back to `Option<T>` for vecs again because there plenty of cases where it matters - not just JSON merge+patch.